pkg/api/util/immutable/immutable.go

```diff
@@ -151,7 +153,15 @@ func validate(path string, v, w reflect.Value, ignoreCase bool) error {
 			}
 			subpath += name
 
-			ic := ignoreCase || strings.EqualFold(v.Type().Field(i).Tag.Get("mutable"), "case")
+			// Read-only properties should be omitted from PUT/POST requests.
+			if strings.EqualFold(structField.Tag.Get("swagger"), "readOnly") {
+				if !v.FieldByIndex([]int{i}).IsZero() {
```

Comment:
This is a bit confusing. I believe this can be changed to 

```suggestion
if !v.Field(i).IsZero() {
```

as this is what happends internally if there is only one value in the array.